tests/jquery.flot.navigate.Test.js

```diff
@@ -378,13 +377,98 @@ describe("flot navigate plugin", function () {
         simulate.sendTouchEvents(finalCoords, placeholder[0].childNodes[2], "touchmove");
         simulate.sendTouchEvents(finalCoords, placeholder[0].childNodes[2], "touchend");
 
-        expect(xaxis.min).toBeCloseTo((midPointCoords.x - initialXmin) * (1 - 1/amount) + initialXmin, 2);
-        expect(xaxis.max).toBeCloseTo(initialXmax - (initialXmax - midPointCoords.x) * (1 - 1/amount), 2);
-        expect(yaxis.min).toBeCloseTo((midPointCoords.y - initialYmin) * (1 - 1/amount) + initialYmin, 2);
-        expect(yaxis.max).toBeCloseTo(initialYmax - (initialYmax - midPointCoords.y) * (1 - 1/amount), 2);
+        expect(xaxis.min).toBeCloseTo((midPointCoords.x - initialXmin) * (1 - 1/amount) + initialXmin, 6);
+        expect(xaxis.max).toBeCloseTo(initialXmax - (initialXmax - midPointCoords.x) * (1 - 1/amount), 6);
+        expect(yaxis.min).toBeCloseTo((midPointCoords.y - initialYmin) * (1 - 1/amount) + initialYmin, 6);
+        expect(yaxis.max).toBeCloseTo(initialYmax - (initialYmax - midPointCoords.y) * (1 - 1/amount), 6);
       });
     });
 
+    describe('touchLogarithmicPan', function() {
+        it('should pan the logarithmic plot', function() {
+            var d1 = [];
+            for (var i = 0; i < 14; i += 0.2) {
+                d1.push([i, 1.01 + Math.sin(i)]);
+            }
+            function piTickGenerator(axis) {
+                var res = [],
+                i = Math.floor(axis.min / Math.PI);
+                do {
+                    var v = i * Math.PI;
+                    res.push([v, i + "\u03c0"]);
+                    ++i;
+                } while (v < axis.max);
+                return res;
+            }
+
+            var plot = $.plot(placeholder, [d1], {
+                series: {
+                    lines: {
+                        show: true
+                    },
+                    points: {
+                        show: true
+                    }
+                },
+                xaxis: {
+                    ticks: piTickGenerator,
+                    autoscale: 'exact'
+                },
+                yaxis: {
+                    mode: 'log',
+                    showTickLabels: "all",
+                    autoscale: 'exact'
+                },
+                zoom: {
+                    interactive: true
+                },
+                pan: {
+                    interactive: true,
+                    enableTouch: true
+                }
+            });
+
+            function getPairOfCoords(x, y) {
```

Comment:
DRY - Define a single getPairOfCoords() outside this function and make all touch tests call it.